src/mapport.cpp

```diff
@@ -125,22 +123,7 @@ static bool ProcessPCP()
 
 static void ThreadMapPort()
 {
-    bool ok;
-    do {
-        ok = false;
-
-        if (g_mapport_enabled_protos & MapPortProtoFlag::PCP) {
-            g_mapport_current_proto = MapPortProtoFlag::PCP;
-            ok = ProcessPCP();
-            if (ok) continue;
-        }
-
-        g_mapport_current_proto = MapPortProtoFlag::NONE;
-        if (g_mapport_enabled_protos == MapPortProtoFlag::NONE) {
-            return;
-        }
-
-    } while (ok || g_mapport_interrupt.sleep_for(PORT_MAPPING_RETRY_PERIOD));
+    while ((!g_mapport_interrupt && ProcessPCP()) || g_mapport_interrupt.sleep_for(PORT_MAPPING_RETRY_PERIOD)) {}
```

Comment:
The thread is not supposed to exit if `ProcessPCP()` fails. It should be more persistent: it should just wait a while and try again, maybe for the user to join a different network, swap their router, upgrade firmware, etc.

Reply:
Yes, this will not exit if  `ProcessPCP` fails. (Unless it's interrupted of course.)

Reply:
Maybe i should make this loop clearer? Like:
```diff
diff --git a/src/mapport.cpp b/src/mapport.cpp
index b303017144..8df78f435d 100644
--- a/src/mapport.cpp
+++ b/src/mapport.cpp
@@ -123,7 +123,9 @@ static bool ProcessPCP()
 
 static void ThreadMapPort()
 {
-    while ((!g_mapport_interrupt && ProcessPCP()) || g_mapport_interrupt.sleep_for(PORT_MAPPING_RETRY_PERIOD)) {}
+    do {
+	ProcessPCP();
+    } while (g_mapport_interrupt.sleep_for(PORT_MAPPING_RETRY_PERIOD)) {}
 }
 
 void StartThreadMapPort()
```

Which also makes clearer that `ProcessPCP()` should actually be `void`.

Reply:
Yes, that seems better. Making ProcessPCP a void is also fine with me.

Reply:
Done.